dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/signatures/KotlinSignatureProvider.kt

```diff
@@ -248,7 +248,7 @@ public class KotlinSignatureProvider(
                 p.modifier[sourceSet].takeIf { it !in ignoredModifiers }?.let {
                         if (it is JavaModifier.Empty) KotlinModifier.Open else it
                     }?.name?.let { keyword("$it ") }
-                p.modifiers()[sourceSet]?.toSignatureString()?.let { keyword(it) }
+                p.modifiers()[sourceSet]?.toSignatureString()?.takeIf { it.isNotEmpty() }?.let { keyword(it) }
```

Comment:
Maybe we can just return `null` from `toSignatureString` (used in 12 places with similar pattern) or even don't add any text in `PageContentBuilder.DocumentableContentBuilder#text` if the text is empty? So to fix all empty spans :)

Reply:
🤷‍♂️ I've just added one more change in place where the `toSignatureString` is not used.

Reply:
Ideally, it should indeed be fixed once and for all, with some generic tests that check we have no empty spans/text at all. 

However, this place and API is very rarely touched, there's a non-zero chance of new rendering bugs or corner cases appearing (especially if changes are made in `DocumentableContentBuilder#text`), and the empty span bugs are very minor and not noticeable by the users.

I'm afraid it might take way more time than what it's worth for now, so I'd focus on just removing obstacles for testing K2 and simplifying the diff.

Reply:
I agree, better not to do it now. Thought, may be we should at least create issue to track this? Something like `Remove empty spans/tags in HTML`? (interesting, are other formatters affected?)

Reply:
I've created that one: https://github.com/Kotlin/dokka/issues/3344
Feel free to rewrite and add possible areas of improvement.